src/eval.jl

```diff
@@ -21,6 +21,28 @@ function Base.iterate(s::StatementsIterator, current_index = 1)
     return (eval_string, new_index)
 end
 
+function stripblock(s)
+    retlines = String[]
+    commonlspace = typemax(Int) 
+    for line in split(s,'\n')
+        if isempty(strip(line))
+            continue
+        end
+        lspace = length(line) - length(lstrip(line))  
+        if lspace < commonlspace
+            commonlspace = lspace
+        end
+    end
+    for line in split(s,'\n')
```

Comment:
i would have to do one pass to determine how much white space to trim right ?

Reply:
the first occurance of whitespace isn't neccesarily the whitespace to trim from all lines

although well formatted code shouldn't look like this , its possible someone is executing something like this:
```julia
for i in 1:10
    dosomething()
  somethingelse()
end
```

Reply:
I mean to avoid doing the `split` twice. You are already colleting in to a vector, might as well collect into it in the first pass.

Reply:
I dont think i understand.
are you suggesting overall only 1 pass is required ?

i could avoid doing the `split` (or `eachline` like you suggested) twice like this :

```julia
function stripblock(s)
    retlines = String[]
    commonlspace = typemax(Int)
    lines = eachline(IOBuffer(s))
    for line in lines
        if isempty(strip(line))
            continue
        end
        lspace = length(line) - length(lstrip(line))  
        if lspace < commonlspace
            commonlspace = lspace
        end
    end
    for line in lines
        if isempty(strip(line))
            continue
        else
            push!(retlines,line[commonlspace+1:end])
        end
    end
    return join(retlines,'\n')
end
```

Reply:
okay i cant do that with `eachline` , because the IOBuffer gets consumed and is empty after the first pass.

Reply:
I meant something like
```julia
function stripblock(s)
    retlines = String[]
    nmin = typemax(Int)
    for line in eachline(IOBuffer(s); keep = true)
        push!(retlines, line)
        if !all(isspace, line)
            # Non-empty line, count number of leading spaces
            n = 0
            for c in line
                c == ' ' || break
                n += 1
            end
            nmin = min(nmin, n)
        end
    end
    # Drop empty leading/trailing lines
    while !isempty(retlines) && all(isspace, retlines[1])
        popfirst!(retlines)
    end
    while !isempty(retlines) && all(isspace, retlines[end])
        pop!(retlines)
    end
    # Drop trailing newline char from the last line
    if !isempty(retlines)
        retlines[end] = chomp(retlines[end])
    end
    # Join while stripping leading whitespace
    io = IOBuffer()
    for line in retlines
        idx = nmin + 1
        if all(isspace, line)
            # Empty line, strip as much leading space as available, but max out at nmin
            i = something(findfirst(x -> x != ' ', line), 1)
            idx = min(i, idx)
        end
        write(io, @view(line[idx:end]))
    end
    return String(take!(io))
end
```

> okay i cant do that with eachline , because the IOBuffer gets consumed and is empty after the first pass.


You can `seekstart` the buffer to go back

Reply:
since you've pretty much rewritten the code would you like to make the PR ?

Reply:
I would prefer if you did . :)